common/achievement/get.ts

```diff
@@ -192,22 +188,31 @@ const assembleAchievementData = async (userAchievements: achievements_with_templ
                               );
                           })
                           .reduce((a, b) => a + b, 0);
+            if (currentValue < maxValue && userAchievements[currentAchievementIndex].achievedAt) {
+                await prisma.user_achievement.update({
+                    where: { id: userAchievements[currentAchievementIndex].id },
+                    data: { achievedAt: null, isSeen: false },
+                });
+            }
         }
     } else {
         currentValue = currentAchievementIndex;
         maxValue = achievementTemplates.length - 1;
     }
 
+    const state: achievement_state = getAchievementState(userAchievements, currentAchievementIndex);
+    const isNewAchievement = state === achievement_state.COMPLETED && !userAchievements[currentAchievementIndex].isSeen;
+
     const achievementContext = transformPrismaJson(
         user,
         userAchievements[currentAchievementIndex].relation,
         userAchievements[currentAchievementIndex].context as Prisma.JsonObject
     );
-    const leftProgress = maxValue - currentValue;
+    const leftProgress = maxValue - currentValue + 1;
```

Comment:
Why +1?

Reply:
E.g.
maxValue (recordValue) is 10
currentValue (currentStreak) is 5
needed to beat the streak is a new record value of 11
-> leftProgress is currentValue + 1